### DISCUSSION TITLE
MPD GUI synchronization

### DISCUSSION THREAD

**Discussion by findnix** [rern/rAudio#851] [Feature Requests]
Hello @rern 

I have built a simple remote control (Zigbee switch -> node red -> mpd) that sends rudimentary MPD commands to rAudio (Vol+, Vol-, fwd, prev, play, stop).
When I control rAudio with the remote control, unfortunately the webgui is not updated or only very late.
Is there any trick or tip and keep the GUI in sync with the MPD status?

Thanks
Ronald

**Comment by rern** [comment]
rAudio runs [Nginx Push Stream Module](https://github.com/wandenberg/nginx-push-stream-module) to broadcast data.
Web user interface subscribes to receive the data. (See [`passive.js`](https://github.com/rern/rAudio-1/blob/6443c51c387bd9e59d8c1b6298c64c1e353f6a32/srv/http/assets/js/passive.js#L86).)

**Comment by findnix** [comment]
Thank You rern!

I took a look at _passive.js_. Now I understand why it's not a good idea to synchronize the GUI directly.
Also, it still happens that reloading an HTML page causes audible glitches, even on an RPi4.

I hereby withdraw my wish :-)

**Reply by rern** [reply]
For the glitches, enable Browser on RPi might help. (Even with no screen connected.)

**Reply by findnix** [reply]
Thank You!
I will test and report :-)
